redbot/cogs/warnings/warnings.py

```diff
@@ -392,27 +392,45 @@ async def warn(
             member = user
         elif isinstance(user, int):
             if not ctx.channel.permissions_for(ctx.guild.me).ban_members:
-                await ctx.send(_("User {user} is not in the server."),format(user=user))
+                await ctx.send(_("I cannot ban users. Missing the permission to ban members."))
```

Comment:
I had the string simply stating that the user was not in the server, because when the `[p]warn` command is used, that is the best response the bot can give. Stating as a response that the bot cannot ban users, without explaining _why_ it is trying to ban is confusing. I think it is better to just state that the user isn't in the server without worrying about the fact that it _could_ have attempted to ban.

Reply:
oh ok, thank you for the clarification. Yes you're right, it would be strange to see that the bot suddenly prompts up with an error that it cannot ban a person when the [p]warn command is evoked. In that case, would you think it makes more sense for the bot to check "if not ctx.channel.permissions_for(ctx.guild.me).ban_members:" when the bot calls the ban() function?

Reply:
No. It makes no sense to ask if the user should be banned if the bot is not going to be able to ban them. A simpler failure message and an even earlier exist is sufficient.